src/reader.rs

```diff
@@ -595,13 +594,9 @@ impl<Term: Terminal> Reader<Term> {
     /// Save history to a the specified file
     ///
     /// # Failure
```

Comment:
One doesn't usually have a `# Failure` header for an `Err(_)` return. Those are typically reserved for cases in which a function may panic. (But the empty line between should be preserved.)

Also, there's a typo on the line above: "to a the" -> "to the".